ietf/meeting/views.py

```diff
@@ -202,32 +202,35 @@ def current_materials(request):
     else:
         raise Http404('No such meeting')
 
+
+def _get_materials_doc(meeting, name):
+    """Get meeting materials document named by name
+
+    Raises Document.DoesNotExist if a match cannot be found.
+    """
+    # try an exact match first
+    doc = Document.objects.filter(name=name).first()
+    if doc is not None and doc.get_related_meeting() == meeting:
+        return doc, None
+    # try parsing a rev number
+    name, rev = name.rsplit('-', 1)
```

Comment:
This isn't sufficient. There are drafts out there with names that are prefixes of other names. You need to make sure the thing after the last - looks like two digits before trying to look up what came before.